plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,457 @@
+		"OPEN",
+		"OPEN_CONFIRM",
+		"OPEN_NOATTR",
+		"OPEN_DOWNGRADE",
+		"CLOSE",
+		"SETATTR",
+		"FSINFO",
+		"RENEW",
+		"SETCLIENTID",
+		"SETCLIENTID_CONFIRM",
+		"LOCK",
+		"LOCKT",
+		"LOCKU",
+		"ACCESS",
+		"GETATTR",
+		"LOOKUP",
+		"LOOKUP_ROOT",
+		"REMOVE",
+		"RENAME",
+		"LINK",
+		"SYMLINK",
+		"CREATE",
+		"PATHCONF",
+		"STATFS",
+		"READLINK",
+		"READDIR",
+		"SERVER_CAPS",
+		"DELEGRETURN",
+		"GETACL",
+		"SETACL",
+		"FS_LOCATIONS",
+		"RELEASE_LOCKOWNER",
+		"SECINFO",
+		"FSID_PRESENT",
+		"EXCHANGE_ID",
+		"CREATE_SESSION",
+		"DESTROY_SESSION",
+		"SEQUENCE",
+		"GET_LEASE_TIME",
+		"RECLAIM_COMPLETE",
+		"LAYOUTGET",
+		"GETDEVICEINFO",
+		"LAYOUTCOMMIT",
+		"LAYOUTRETURN",
+		"SECINFO_NO_NAME",
+		"TEST_STATEID",
+		"FREE_STATEID",
+		"GETDEVICELIST",
+		"BIND_CONN_TO_SESSION",
+		"DESTROY_CLIENTID",
+		"SEEK",
+		"ALLOCATE",
+		"DEALLOCATE",
+		"LAYOUTSTATS",
+		"CLONE",
+		"COPY",
+		"OFFLOAD_CANCEL",
+		"LOOKUPP",
+		"LAYOUTERROR",
+		"COPY_NOTIFY",
+		"GETXATTR",
+		"SETXATTR",
+		"LISTXATTRS",
+		"REMOVEXATTR",
+	}
+
+	nfs3Ops = make(map[string]bool)
+	nfs4Ops = make(map[string]bool)
+
+	if len(n.IncludeOperations) == 0 {
+		for _, Op := range nfs3Fields {
+			nfs3Ops[Op] = true
+		}
+		for _, Op := range nfs4Fields {
+			nfs4Ops[Op] = true
+		}
+	} else {
+		for _, Op := range n.IncludeOperations {
+			nfs3Ops[Op] = true
+		}
+		for _, Op := range n.IncludeOperations {
+			nfs4Ops[Op] = true
+		}
+	}
+
+	if len(n.ExcludeOperations) > 0 {
+		for _, Op := range n.ExcludeOperations {
+			if nfs3Ops[Op] {
+				delete(nfs3Ops, Op)
+			}
+			if nfs4Ops[Op] {
+				delete(nfs4Ops, Op)
+			}
+		}
+	}
+
+	for scanner.Scan() {
+		line := strings.Fields(scanner.Text())
+
+		if len(line) == 0 {
+			break
+		}
+
+		if in(line, "fstype") && (in(line, "nfs") || in(line, "nfs4")) && len(line) > 4 {
+			device = line[4]
+			export = line[1]
+			skip = false
+		} else if (in(line, "(nfs)") || in(line, "(nfs4)")) && len(line) > 5 {
+			version = strings.Split(line[5], "/")[1]
+		}
+
+		if len(n.IncludeMounts) > 0 {
+			skip = true
+			for _, RE := range n.IncludeMounts {
+				matched, _ := regexp.MatchString(RE, device)
+				if matched {
+					skip = false
+					break
+				}
+			}
+		}
+
+		if !skip && len(n.ExcludeMounts) > 0 {
+			for _, RE := range n.ExcludeMounts {
+				matched, _ := regexp.MatchString(RE, device)
+				if matched {
+					skip = true
+					break
+				}
+			}
+		}
+
+		if !skip {
+			n.parseStat(device, export, version, line, n.Fullstat, nfs3Ops, nfs4Ops, acc)
+		}
+	}
+	return nil
+}
+
+func (n *NFSClient) getMountStatsPath() string {
+
+	path := "/proc/self/mountstats"
+	if os.Getenv("MOUNT_PROC") != "" {
+		path = os.Getenv("MOUNT_PROC")
+	}
+	n.Log.Debugf("Opening [%s] for mountstats", path)
+	return path
+}
+
+func (n *NFSClient) Gather(acc telegraf.Accumulator) error {
+
+	file, err := os.Open(n.getMountStatsPath())
+	if err != nil {
+		n.Log.Errorf("Failed opening the /proc/self/mountstats file:  ", err)
```

Comment:
This should also output the actual path in case it is set from environment.

Reply:
Fair point.  Fixed